wire-ios-utilities/Source/UTIHelper.swift

```diff
@@ -224,54 +192,44 @@ public final class UTIHelper: NSObject {
 
     @objc
     public class func convertToUti(mime: String) -> String? {
-        if #available(iOS 14, *) {
-            var utType: UTType? = UTType(mimeType: mime)
+        var utType: UTType? = UTType(mimeType: mime)
 
-            #if targetEnvironment(simulator)
-            /// HACK: hard code MIME when preferredMIMEType is nil for M1 simulator, we should file a ticket to apple for this issue
-            if utType == nil {
-                utType = createUTType(mime: mime)
-            }
-            #endif
-
-            return utType?.identifier
-        } else {
-            return UTTypeCreatePreferredIdentifierForTag(kUTTagClassMIMEType,
-                                                         mime as CFString,
-                                                         nil)?.takeRetainedValue() as String?
+        #if targetEnvironment(simulator)
+        /// HACK: hard code MIME when preferredMIMEType is nil for M1 simulator, we should file a ticket to apple for this issue
+        if utType == nil {
+            utType = createUTType(mime: mime)
         }
+        #endif
+
+        return utType?.identifier
     }
 
     public class func convertToMime(fileExtension: String) -> String? {
-        if #available(iOS 14, *) {
-            var utType: UTType? = UTType(filenameExtension: fileExtension)
-
-            // for uttype not conforming data, e.g com.apple.pkpass, retry with conformingTo: nil
-            if utType == nil {
-                utType = UTType(tag: fileExtension, tagClass: .filenameExtension, conformingTo: nil)
-            }
+        var utType: UTType? = UTType(filenameExtension: fileExtension)
 
-            #if targetEnvironment(simulator)
-            /// HACK: hard code MIME when preferredMIMEType is nil for M1 simulator, we should file a ticket to apple for this issue
-            if utType == nil {
-                utType = utTypes.first(where: { $0.fileExtension == fileExtension })
-            }
-            #endif
+        // for uttype not conforming data, e.g com.apple.pkpass, retry with conformingTo: nil
+        if utType == nil {
+            utType = UTType(tag: fileExtension, tagClass: .filenameExtension, conformingTo: nil)
+        }
 
-            var mimeType: String?
-            if let utType = utType {
-                mimeType = mime(from: utType)
-            }
+        #if targetEnvironment(simulator)
+        /// HACK: hard code MIME when preferredMIMEType is nil for M1 simulator, we should file a ticket to apple for this issue
```

Comment:
is it really an issue for M1 simulator ? I found https://developer.apple.com/forums/thread/685968

Reply:
removed